tests/test_rio_blocks.py

```diff
@@ -0,0 +1,132 @@
+"""Unittests for $ rio blocks"""
+
+
+import itertools as it
+import json
+import os
+import re
+
+import numpy as np
+import six
+
+import rasterio
+from rasterio.warp import transform_bounds
+from rasterio.rio.main import main_group
+
+
+def check_features_block_windows(features, src, bidx):
+
+    """Compare GeoJSON features to a datasource's blocks + windows.
+
+    Parameters
+    ----------
+    features : iter
+        GeoJSON features.
+    src : RasterReader
+        Open input datasource.
+    """
+
+    out = []
+    iterator = six.moves.zip_longest(features, src.block_windows(bidx=bidx))
+    for feat, (block, window) in iterator:
+
+        out.append(np.array_equal(
+            json.loads(feat['properties']['block']),
+            block))
+
+        out.append(np.array_equal(
+            json.loads(feat['properties']['window']),
+            window))
+
+    return all(out)
+
+
+def test_windows(runner, path_rgb_byte_tif):
```

Comment:
@geowurster I think you should also test for the coordinates being close to expected values in this function and in `test_windows_projected` below.  Just testing one feature's coordinates would be sufficient.